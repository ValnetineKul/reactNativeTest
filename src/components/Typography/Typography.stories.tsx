import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";

import { Typography } from "./Typography";
import { View } from "react-native";

export default {
  title: "components/Typography",
  component: Typography,
} as ComponentMeta<typeof Typography>;

export const Basic: ComponentStory<typeof Typography> = () => {
  return (
    <View style={{ backgroundColor: "lightgray" }}>
      <Typography color="black">Hello world</Typography>
      <Typography color="blue">Hello world</Typography>
      <Typography color="error">Hello world</Typography>
      <Typography color="gray">Hello world</Typography>
      <Typography color="white">Hello world</Typography>
    </View>
  );
};
