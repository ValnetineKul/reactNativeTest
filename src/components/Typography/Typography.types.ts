export enum FontWeights {
  fontWeightThin = "100",
  fontWeightExtraLight = "200",
  fontWeightLight = "300",
  fontWeightRegular = "400",
  fontWeightMedium = "500",
  fontWeightSemiBold = "600",
  fontWeightBold = "700",
  fontWeightExtraBold = "800",
  fontWeightHeavy = "900",
}

export type TypographyVariants =
  | "h1"
  | "h2"
  | "h3"
  | "h4"
  | "h5"
  | "h6"
  | "body1"
  | "body2"
  | "caption";

export type TypographyColors = "black" | "white" | "gray" | "blue" | "error";
