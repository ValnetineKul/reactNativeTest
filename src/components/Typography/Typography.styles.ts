import { StyleSheet } from "react-native";
import {
  COLOR_ERROR,
  COLOR_TEXT_BLACK,
  COLOR_TEXT_BLUE,
  COLOR_TEXT_GRAY,
  COLOR_TEXT_WHITE,
} from "../../theme";
import { FontWeights } from "./Typography.types";

export const styles = StyleSheet.create({
  font: {
    fontFamily: "Roboto",
  },
  h1: {
    fontWeight: FontWeights.fontWeightBold,
    fontSize: 38,
    lineHeight: 50,
  },
  h2: {},
  h3: {},
  h4: {},
  h5: {},
  h6: {
    fontWeight: FontWeights.fontWeightMedium,
    fontSize: 20,
    lineHeight: 24,
    letterSpacing: 0.15,
  },
  body1: {
    fontWeight: FontWeights.fontWeightRegular,
    fontSize: 15,
    lineHeight: 20,
  },
  body2: {
    fontWeight: FontWeights.fontWeightBold,
    fontSize: 15,
    lineHeight: 20,
  },
  caption: {
    fontWeight: FontWeights.fontWeightRegular,
    fontSize: 12,
    lineHeight: 16,
  },
  white: {
    color: COLOR_TEXT_WHITE,
  },
  black: {
    color: COLOR_TEXT_BLACK,
  },
  gray: {
    color: COLOR_TEXT_GRAY,
  },
  blue: {
    color: COLOR_TEXT_BLUE,
  },
  error: {
    color: COLOR_ERROR,
  },
});
