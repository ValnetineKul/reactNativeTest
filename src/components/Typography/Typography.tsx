import React, { type PropsWithChildren } from "react";
import { Text, TextStyle } from "react-native";
import { styles } from "./Typography.styles";
import { TypographyColors, TypographyVariants } from "./Typography.types";

type TypographyProps = {
  variant?: TypographyVariants;
  color?: TypographyColors;
  style?: TextStyle | TextStyle[];
};

export const Typography = ({
  variant = "body1",
  color = "black",
  children,
  style,
}: PropsWithChildren<TypographyProps>) => {
  return (
    <Text style={[styles.font, styles[variant], styles[color], style]}>
      {children}
    </Text>
  );
};
