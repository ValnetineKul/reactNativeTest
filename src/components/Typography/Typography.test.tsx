import React from "react";
import renderer from 'react-test-renderer';
import { render, screen } from "@testing-library/react-native";
import { Typography } from "./Typography";

describe("Typography tests", () => {
  it("renders the correct text", () => {
    render(<Typography>Hello world</Typography>);
    expect(screen.getByText("Hello world")).toBeTruthy();
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Typography>Hello world</Typography>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
