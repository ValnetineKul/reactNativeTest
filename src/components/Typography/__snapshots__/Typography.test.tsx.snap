// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography tests renders correctly 1`] = `
<Text
  style={
    [
      {
        "fontFamily": "Roboto",
      },
      {
        "fontSize": 15,
        "fontWeight": "400",
        "lineHeight": 20,
      },
      {
        "color": "#4A4A4A",
      },
      undefined,
    ]
  }
>
  Hello world
</Text>
`;
